### ISSUE TITLE
Linux build not working 

### ISSUE THREAD

**Issue by ErichZimmer** [ErichZimmer/OpenPIV-Python-cxx#19]
**Describe the bug**
When the GitHub CI runs the  cmake  configuration step in `_build_utilities` (e.g. `cmake -S . -B build`), the error `Cannot specify link libraries for target "Change Dir:  /home/runner/work/OpenPIV-Python-cxx/OpenPIV-Python-cxx/extern/openpiv-c--qt/build/CMakeFiles/CMakeTmp` occurs.

**To Reproduce**
1. Go to GitHub actions or clone this repository 
2. Click on Linux build or type `python setup.py build`
3. Scroll down to line 303 on the output from cmake 

**Expected behavior**
The library builds like the Windows and macOS tests.

**Additional context**
Due to lack of experience in programming and mental fogginess from working 55 hours a week with 2 hour commutes, I need some help on this issue. Obviously, my very naive approaches are not working (doesn't help that this is my first true package). Perhaps, @timdewhirst or @alexlib can help?

**Comment by alexlib** [comment]
No idea. maybe to ask `scikit-build` guys?

**Comment by timdewhirst** [comment]
I should be able to take a look, probably not until closer to the weekend however!

It might also be useful to look at https://github.com/OpenPIV/openpiv-c--qt/tree/15-add-pybind11-bindings-for-core-classes - I've been (slowly) adding python bindings to the core classes, and have tried to implement pip/setup.py based building. It's not quite there yet - it doesn't clean up properly and subsequent builds fail as a result - but might be a helpful start?

**Comment by ErichZimmer** [comment]
@timdewhirst 
On cleaning up after builds, I added a check in case there was a previous build in my setup.py as I encountered the same issue. 
I looked at your branch and it is helpful as an additional example of wrapping classes (when I get to tests based on solely the FFTs themselves).

@alexlib 
I will wait until after the weekend when I should have a few hours that I can concentrate on this issue. If not much occurs then, then I will head your suggestion.

**Comment by ErichZimmer** [comment]
I am a little puzzled on why the Linux build tests fails while the `openpiv-c--qt` repository passed all build tests on main branch.

**Comment by ErichZimmer** [comment]
It is interesting to note that readthedocs encounters the same cmake errors as the CI builds.
https://readthedocs.org/projects/openpiv-python-cxx/builds/18056120/

**Comment by alexlib** [comment]
@ErichZimmer  the folder is missing indeed:

      FileNotFoundError: [Errno 2] No such file or directory: '/workspaces/OpenPIV-Python-cxx/openpiv-python-cxx/extern/openpiv-c--qt/external/vcpkg/packages'

please check the way `git clone --recursive` gets the stuff - on Linux there's no `vcpkg\packages`, only these subfolders 

![image](https://user-images.githubusercontent.com/747110/190642206-42f2b2d9-3652-4445-b4bc-d7f3cc6f8190.png)

**Comment by alexlib** [comment]
and there is no such a folder here: https://github.com/microsoft/vcpkg/tree/17f96c51ab9ae6d27bdb5cdea3fe84ac71cc2de1

**Comment by alexlib** [comment]
so when this line was born and on what system? 
set(PKG_PATHS ${OPENPIV_CXX_DIR}/external/vcpkg/packages)
I left you a comment in the CMakeLists.txt

**Comment by ErichZimmer** [comment]
@alexlib I redone the cmake build system, but there appears to be some unreferenced variables causing some issues.

**Comment by alexlib** [comment]
> @alexlib I redone the cmake build system, but there appears to be some unreferenced variables causing some issues.

@timdewhirst could you please take a look - I'm not an expert on `cmake` or anything. It's strange that the compilation works on Windows and Mac OS X but not on Linux.

**Comment by alexlib** [comment]
It works for me now 
```
Using /home/user/.edm/envs/openpivcxx/lib/python3.8/site-packages
Finished processing dependencies for OpenPIV-cxx==0.2.4
```

**Comment by alexlib** [comment]
@ErichZimmer looking through the compilation that worked for me and comparing to the github action I see that my compilation didn't include Ninja part at all. @timdewhirst do you know something about it?

**Comment by alexlib** [comment]
@ErichZimmer it was simple - just merge the pull-request. I've verified on my fork and `ubuntu-latest` works fine

**Comment by ErichZimmer** [comment]
Fixed with #20
